src/one/util/ideaplugin/screenshoter/SaveImageAction.java

```diff
@@ -0,0 +1,69 @@
+package one.util.ideaplugin.screenshoter;
+
+import com.intellij.notification.NotificationDisplayType;
+import com.intellij.notification.NotificationGroup;
+import com.intellij.openapi.actionSystem.*;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.MessageType;
+import com.intellij.openapi.util.io.FileUtil;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.util.SystemProperties;
+import org.jetbrains.annotations.NotNull;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+
+public class SaveImageAction extends AnAction {
+
+    private static final NotificationGroup NOTIFICATION_GROUP = new NotificationGroup(
+            "image.saved.id",
+            NotificationDisplayType.STICKY_BALLOON,
+            false,
+            null
+    );
+
+    @Override
+    public void actionPerformed(AnActionEvent anActionEvent) {
+        Editor editor = CopyImageAction.geEditor(anActionEvent);
+        if (editor == null) return;
+
+        BufferedImage image = new ImageBuilder(editor).createImage();
+        saveImage(editor, image);
+    }
+
+
+    private void saveImage(@NotNull Editor editor, @NotNull BufferedImage image) {
+        Project project = editor.getProject();
+        CopyImageOptionsProvider.State options = CopyImageOptionsProvider.getInstance(project).getState();
+        String toSave = options.myDirectoryToSave;
+        if (StringUtil.isEmpty(toSave)) {
+            toSave = SystemProperties.getUserHome();
+        }
+        try {
+            SimpleDateFormat dt = new SimpleDateFormat("yyyyymmdd_hhmmss");
+            String date = dt.format(new Date());
```

Comment:
I'm pretty sure `yyyyymmdd_hhmmss` pattern is invalid: `mm` cannot encode both months and minutes. Also `h` is am/pm-based (1-12), but you don't add a suffix. Should be like `yyyyMMdd_HHmmss`. Also why not modern `DateTimeFormatter` and `LocalDateTime.now()`?

Reply:
thx, actually it is not important (it just provides more or less unique name) but I will fix it